cpp/src/binary/jit/core/launcher.cpp

```diff
@@ -62,35 +51,47 @@ namespace jit {
         return nullptr;
     }
 
+    Launcher& Launcher::setProgram(std::string prog_file_name)
+    {
+        auto startPointClock = std::chrono::high_resolution_clock::now();
+
+            program = cacheInstance.getProgram(prog_file_name,
+                                                code::kernel,
+                                                headersName,
+                                                compilerFlags,
+                                                headersCode);
+
+        auto stopPointClock = std::chrono::high_resolution_clock::now();
+        std::chrono::duration<double> elapsed_seconds = stopPointClock-startPointClock;
+        std::cout << "Program Deserialize (ms): " << elapsed_seconds.count()*1000 << std::endl;
+    }
+
     Launcher::Launcher()
-     : program {JitCache.program(code::kernel, headersName, compilerFlags, headersCode)}
-    { }
+     : cacheInstance{cudf::jit::cudfJitCache::Instance()}
+    { 
+        this->setProgram("prog_binop");
+    }
 
     Launcher::Launcher(Launcher&& launcher)
      : program {std::move(launcher.program)}
+     , cacheInstance {cudf::jit::cudfJitCache::Instance()}
+     , kernel_inst {std::move(launcher.kernel_inst)}
     { }
 
-    Launcher& Launcher::kernel(std::string&& value) {
-        kernelName = value;
-        return *this;
-    }
-
     gdf_error Launcher::launch(gdf_column* out, gdf_column* lhs, gdf_scalar* rhs) {
-        program.kernel(kernelName.c_str())
-               .instantiate(arguments)
-               .configure_1d_max_occupancy()
-               .launch(out->size,
-                       out->data, lhs->data, rhs->data);
+
+        getKernel().configure_1d_max_occupancy()
```

Comment:
Eventually we will want to have a way to launch with an explicit configuration of blocks and threads, won't we?

Reply:
The launcher is binop specific. The cache is generic. I spent quite some thought on how to make the launcher generic and realized that it doesn't do that much outside binop specific work.